src/dbt_client/dbtClientFactory.ts

```diff
@@ -0,0 +1,23 @@
+import { SourceFileChangedEvent } from "../manifest/sourceFileChangedEvent";
+import { getPythonPathFromExtention } from "../utils";
+import { dbtClient } from "./dbtClient";
+
+export class DBTClientFactory {
+    static async createDBTClient() {
```

Comment:
I would move the instantiation logic of this to the DBTProjectContainer. That way other code only needs to integrate with that. If there is a DBTClient on the DBTProjectContainer, that means that you have a pythonPath. That way you can also get rid of the singleton integration of DBTClient, because in that way you have to support a DBTClient without a pythonPath.